cpp/src/arrow/adapters/orc/adapter_test.cc

```diff
@@ -536,6 +536,15 @@ TEST_F(TestORCWriterTrivialNoConversion, writeFilledChunkAndSelectField) {
                             &selected_indices);
 }
 
+TEST_F(TestORCWriterTrivialNoConversion, writeSlicedBatch) {
+  std::shared_ptr<Table> table =
+      GenerateRandomTable(table_schema, /*size=*/100, /*min_num_chunks=*/1,
```

Comment:
Does it cover type like `list<list<X>>` to make sure that offset of the inner list is correct?

Reply:
The test doesn't cover that case. But it should be fine, since `NormalizeArray` is recursive. So  `NormalizeArray(list<list<X>>)` calls `NormalizeArray(list<X>)`, which calls `NormalizeArray(X)`. The offsets at each level should be handled.